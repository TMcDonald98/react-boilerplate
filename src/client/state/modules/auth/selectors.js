
export const getApiKey = state => state?.auth?.apiKey;
