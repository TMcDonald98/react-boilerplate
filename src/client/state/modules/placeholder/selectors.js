
export const getPlaceholder = state => state?.placeholder?.data;
export const getApiKey = state => state?.app?.apiKey;
