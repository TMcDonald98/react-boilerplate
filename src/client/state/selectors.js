
export * from './modules/auth/selectors';
export * from './modules/location/selectors';
export * from './modules/placeholder/selectors';
