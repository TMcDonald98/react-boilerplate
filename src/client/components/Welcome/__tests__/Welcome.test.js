import * as React from 'react';
import { getLocationType } from 'state/modules/location/selectors';
import {
  Welcome,
  mapStateToProps
} from '../Welcome';

jest.mock('state/modules/location/selectors');

describe('components/Welcome', () => {
  describe('render', () => {
    it('should render without error', () => {
      const result = shallow(<Welcome />);

      expect(result).toMatchSnapshot('Welcome');
    });
  });

  describe('class methods', () => {

    const changeScreen = () => {
      const result = shallow(<Welcome />);

      expect(result.state('screen')).toEqual(result.instance().defaultScreen);

      const testScreen = 'yes';
      result.instance().changeScreen(testScreen)();
      expect(result.state('screen')).toEqual(testScreen);
      return result;
    };

    describe('changeScreen', () => {
      it('should change the screen state', () => {
        changeScreen();
      });
    });

    describe('changeToDefaultScreen', () => {
      it('should change the screen back to the default state', () => {
        const result = changeScreen();

        result.instance().changeToDefaultScreen();
        expect(result.state('screen')).toEqual(result.instance().defaultScreen);
      });
    });
  });
});
