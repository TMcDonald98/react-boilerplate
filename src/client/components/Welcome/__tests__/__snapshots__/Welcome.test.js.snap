// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Welcome render should render without error: Welcome 1`] = `
<div
  className="welcome-container"
>
  <CSSTransition
    appear={true}
    classNames="fade-up"
    in={true}
    timeout={300}
  >
    <div
      className="welcome-screen welcome-screen_start"
    >
      <h1
        className="welcome-title"
      >
        Hello World!
      </h1>
      <p
        className="welcome-cta"
      >
        Click one of these buttons!
      </p>
      <div
        className="welcome-button-container"
      >
        <button
          className="button-success"
          onClick={[Function]}
        >
          Yes!
        </button>
        <button
          className="button-failure"
          onClick={[Function]}
        >
          No!
        </button>
      </div>
    </div>
  </CSSTransition>
  <CSSTransition
    appear={true}
    classNames="fade-in"
    in={false}
    timeout={
      Object {
        "enter": 500,
        "exit": 300,
      }
    }
    unmountOnExit={true}
  >
    <div
      className="welcome-screen welcome-screen_yes"
    >
      <h1>
        Great!
      </h1>
      <Check />
    </div>
  </CSSTransition>
  <CSSTransition
    appear={true}
    classNames="fade-in"
    in={false}
    timeout={
      Object {
        "enter": 500,
        "exit": 300,
      }
    }
    unmountOnExit={true}
  >
    <div
      className="welcome-screen welcome-screen_no"
    >
      <h1>
        Unfortunate!
      </h1>
      <X />
    </div>
  </CSSTransition>
</div>
`;
